src/dialog/src/Dialog.js

```diff
@@ -299,11 +306,15 @@ class Dialog extends React.Component {
                 <Heading is="h4" size={600} flex="1">
                   {title}
                 </Heading>
-                <IconButton
-                  appearance="minimal"
-                  icon="cross"
-                  onClick={() => onCancel(close)}
-                />
+                {hasClose ? (
+                  <IconButton
+                    appearance="minimal"
+                    icon="cross"
+                    onClick={() => onCancel(close)}
+                  />
+                ) : (
+                  <IconButton appearance="minimal" icon="blank" />
```

Comment:
If we don't add blank icon, the header without title collapses

Reply:
@mshwery we can keep that collapsed if that's ok

Reply:
@mshwery ok I've removed icon if `hasClose` is false. I guess blank title without close icon is edge-case.

Reply:
Could you provide some screenshots of this behavior (and the of change overall)?

Reply:
Sure.
Dialog without close:
![image](https://user-images.githubusercontent.com/300067/54694842-49fdda80-4aff-11e9-8e36-451d2c78ed14.png)
Corner dialog without close:
![image](https://user-images.githubusercontent.com/300067/54694872-56823300-4aff-11e9-9226-3175cf7250a7.png)

Reply:
Nice! What does it look like when the title collapses like you were describing (without the placeholder empty icon)?